packages/engine/src/parseErrors.ts

```diff
@@ -51,60 +78,64 @@ export interface ErrorInfo extends ErrorInfoWithoutImprovedError {
 
 export interface ParseErrorsOpts {}
 
-export const parseErrors = (
-  message: string,
-): ErrorInfoWithoutImprovedError[] => {
+export const parseErrorsWithDb = (db: TsErrorMessageDb, message: string) => {
```

Comment:
Refactored this so we can test it with a custom db, cleaning up tests and making them more robust for when TS changes its error messages.